core/util_test.go

```diff
@@ -170,11 +170,12 @@ func BatchNewDrand(t *testing.T, n int, insecure bool, sch scheme.Scheme, beacon
 		daemons[i] = daemon
 		drands[i] = bp
 
-		// to make sure to stop all daemon after each test
+		// Make sure to stop all daemon after each test
 		t.Cleanup(func() {
 			ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+			defer cancel()
 			daemon.Stop(ctx)
-			cancel()
+			time.Sleep(3 * time.Second)
```

Comment:
is this to wait for all the goroutines to exit?

Reply:
That's correct.